app/components/modal-nfc/modal-nfc.tsx

```diff
@@ -122,38 +146,42 @@ export const ModalNfc: React.FC<{
       })
       logParseDestinationResult(destination)
 
-      if (destination.valid) {
+      if (destination.valid && settlementAmount && convertMoneyAmount) {
         if (destination.destinationDirection === DestinationDirection.Send) {
           Alert.alert(LL.SettingsScreen.nfcOnlyReceive())
         } else {
-          navigation.reset({
-            routes: [
-              {
-                name: "Primary",
-              },
-              {
-                name: "redeemBitcoinDetail",
-                params: {
-                  receiveDestination: destination,
-                },
-              },
-            ],
-          })
+          let amount = settlementAmount.amount
+          if (settlementAmount.currency === WalletCurrency.Usd) {
+            amount = convertMoneyAmount(
+              toUsdMoneyAmount(settlementAmount.amount),
+              WalletCurrency.Btc,
+            ).amount
+          }
+
+          destination.validDestination.minWithdrawable = amount * 1000 // coz msats
+          destination.validDestination.maxWithdrawable = amount * 1000 // coz msats
+
+          console.log(destination)
+
+          receiveViaNFC(destination, toBtcMoneyAmount(settlementAmount.amount))
         }
       }
 
       dismiss()
     }
 
     init()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
can you add a comment on why this is necessary? this has been a source of bug in the past